### PR TITLE
Delete unused Crossgen2PackageRID property

### PR DESCRIPTION
**Pull request by am11**
Unused after https://github.com/dotnet/runtime/pull/65948

### CODE REVIEW

**Review by mangod9 — APPROVED** [review]
Thanks for fixing!

**Comment by mangod9** [comment]
I managed to get one dump so far and we are continuing to investigate. Current theory is that this is somehow related to a field within `System.Runtime.CompilerServices.AsyncTaskMethodBuilder1+AsyncStateMachineBox` getting corrupted. 

Otherwise running `build -test` still fails with this error, assume this is an orthogonal failure though?  

```
System.IO.IOException : The process cannot access the file '\\\\?\\D:\\git\\installer\\artifacts\\bin\\EndToEnd.Tests\\Debug\\net7.0\\Tests\\EndToEnd.Tests\\ItDoesNotRollForwardToTheLatestVersionMicrosoft.AspNetCore.All_2.1\\TestAppSimple' because it is being used by another process.
   at System.IO.FileSystem.RemoveDirectoryInternal(String fullPath, Boolean topLevel, Boolean allowDirectoryNotEmpty)
   at System.IO.FileSystem.RemoveDirectoryRecursive(String fullPath, WIN32_FIND_DATA& findData, Boolean topLevel)
   at System.IO.FileSystem.RemoveDirectory(String fullPath, Boolean recursive)
   at System.IO.DirectoryInfo.Delete(Boolean recursive)
   at Microsoft.DotNet.TestFramework.DirectoryInfoExtensions.EnsureExistsAndEmpty(DirectoryInfo subject) in D:\git\installer\test\Microsoft.DotNet.Tools.Tests.Utilities\Extensions\DirectoryInfoExtensions.cs:line 29
   at Microsoft.DotNet.TestFramework.TestAssets.CreateTestDirectory(String testProjectName, String callingMethod, String identifier) in D:\git\installer\test\Microsoft.DotNet.Tools.Tests.Utilities\TestAssets.cs:line 73
   at Microsoft.DotNet.TestFramework.TestAssetInfo.GetTestDestinationDirectory(String callingMethod, String identifier) in D:\git\installer\test\Microsoft.DotNet.Tools.Tests.Utilities\TestAssetInfo.cs:line 69
   at Microsoft.DotNet.TestFramework.TestAssetInfo.CreateInstance(String callingMethod, String identifier) in D:\git\installer\test\Microsoft.DotNet.Tools.Tests.Utilities\TestAssetInfo.cs:line 52
   at EndToEnd.TestProjectCreator.Create() in D:\git\installer\test\EndToEnd\TestProjectCreator.cs:line 38
   at EndToEnd.GivenFrameworkDependentApps.ItDoesNotRollForwardToTheLatestVersion(String packageName, String minorVersion) in D:\git\installer\test\EndToEnd\GivenFrameworkDependentApps.cs:line 56
   at EndToEnd.GivenFrameworkDependentApps.ItDoesNotRollForwardToTheLatestVersionOfAspNetCoreAll(String minorVersion) in D:\git\installer\test\EndToEnd\GivenFrameworkDependentApps.cs:line 45
```

**Comment by am11** [comment]
Remaining CI failures seem unrelated infra issues.

---

#### Regarding second commit
cc @wfurt, @Thefrank

When building natively on FreeBSD, proper crossgen2 binary is produced (i.e. the one with FreeBSD compatible apphost). 

However, when cross-compiling for FreeBSD from Linux, we had PublishReadyToRun disabled, but we were still producing shipping package with invalid CG2 binary (with linux apphost...). First commit uncovers that issue at package-restore-time and second commit prevents CG2 package generation in FreeBSD cross-compile mode.

File: Directory.Build.props [hunk 1]
```diff
@@ -210,6 +210,12 @@
   </PropertyGroup>
 
   <PropertyGroup>
+    <!-- Disable crossgen on:
+          * NetBSD, illumos and Solaris for now. This can be revisited when we have full support.
+          * FreeBSD when cross building from Linux. -->
+    <DisableCrossgen2 Condition="'$(TargetOS)'=='NetBSD' or '$(TargetOS)'=='illumos' or '$(TargetOS)'=='Solaris' or
+        ('$(TargetOS)'=='FreeBSD' and '$(CrossBuild)'=='true')">true</DisableCrossgen2>
```
Comment (am11) [inline Directory.Build.props:217]: Opened #66866 to use live-build of apphost on these platforms.

cc @ViktorHofer, @agocke, ideally we should _always_ use the live apphost for shipping tools, unless there is an issue with that?

**Comment by Thefrank** [comment]
@am11 so crossbuilding FreeBSD under Linux was trying to restore a package that didn't exist (FreeBSD Crossgen2)? If yes, then I think your https://github.com/dotnet/runtime/issues/66866 would be a good way of allowing community supported platforms to produce a "more complete" runtime or full SDK package via crossbuilding.

**Comment by am11** [comment]
@Thefrank, correct. The apphost package (`microsoft.netcore.app.crossgen2.<platform>-<arch>.<version>.nupkg`) restored for crossgen2 on the `main` branch is using linux-x64 when we pass `-cross` or `-p:CrossBuild=true` to `./build.sh`. Which is incorrect when target is anything other than linux-x64. The crossgen2 tool used to cross build _during_ the build is correct (i.e. when `CrossBuild = true`, it should use CG2 built with host compatible apphost, rather than that of target's).

The reason why we don't see any error on main branch is because of the wrong `Crossgen2PackageRID` value which is fixed by first commit in this PR. With that fix, the FreeBSD CI leg started to fail during apphost package restore like this:

> .dotnet/sdk/6.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(129,5): error NETSDK1084: (NETCORE_ENGINEERING_TELEMETRY=Restore) There is no application host available for the specified RuntimeIdentifier 'freebsd-x64'.

The second commit disables crossgen2 project build on FreeBSD package generation in cross build mode. It was already partially disabled, but not properly excluded since the issue was hidden due to the said bug. When #66866 is resolved, we can enable it back.

**Comment by am11** [comment]
FreeBSD issue is handled by https://github.com/dotnet/runtime/pull/65948 merge (we do not restore apphost for CG2 on unsupported platforms anymore). I have reverted the second commit and merged `main` to my PR branch.

Libs test failures are unrelated to this change.

**Comment by mangod9** [comment]
Hi @am11, assume this is ok to merge?

**Review by ViktorHofer — APPROVED** [review]

**Comment by am11** [comment]
This was folded into https://github.com/dotnet/runtime/pull/67636.

**Comment by mangod9** [comment]
thanks for the update.

**Pull request closed without merging**